packages/web-api-scan-request-sender/src/sender/on-demand-scan-request-sender.ts

```diff
@@ -19,13 +19,19 @@ export class OnDemandScanRequestSender {
             onDemandPageScanRequests.map(async page => {
                 const resultDocs = await this.onDemandPageScanRunResultProvider.readScanRuns([page.id]);
                 const resultDoc = resultDocs.pop();
+                let isEnqueueSuccessful: boolean;
                 if (resultDoc !== undefined && resultDoc.run !== undefined && resultDoc.run.state === 'accepted') {
                     const message = this.createOnDemandScanRequestMessage(page);
-                    await this.queue.createMessage(this.storageConfig.scanQueue, message);
+                    isEnqueueSuccessful = await this.queue.createMessage(this.storageConfig.scanQueue, message);
+                }
+
+                if (isEnqueueSuccessful === true) {
                     await this.updateOnDemandPageResultDoc(resultDoc);
                 }
 
-                await this.pageScanRequestProvider.deleteRequests([page.id]);
+                if (isEnqueueSuccessful !== false) {
+                    await this.pageScanRequestProvider.deleteRequests([page.id]);
```

Comment:
this could be in the previous if block

Reply:
Correct me if I'm wrong, if the value is undefined we still need to delete message (I think this is to delete requests that has state not equal to 'accepted'). This is the current behavior.